src/libraries/System.Net.NameResolution/src/System/Net/Dns.cs

```diff
@@ -641,9 +641,7 @@ private static bool LogFailure(object hostNameOrAddress, long? startingTimestamp
         /// Since this is doing synchronous work on a thread pool thread, we want to limit how many threads end up being
         /// blocked.  We could employ a semaphore to limit overall usage, but a common case is that DNS requests are made
         /// for only a handful of endpoints, and a reasonable compromise is to ensure that requests for a given host are
-        /// serialized.  Once the data for that host is cached locally by the OS, the subsequent requests should all complete
-        /// very quickly, and if the head-of-line request is taking a long time due to the connection to the server, we won't
-        /// block lots of threads all getting data for that one host.  We also still want to issue the request to the OS, rather
```

Comment:
> Once the data for that host is cached locally by the OS, the subsequent requests should all complete very quickly, and if the head-of-line request is taking a long time due to the connection to the server, we won't block lots of threads all getting data for that one host

Essentially, this turned out to be a false assumption. In Ubuntu there is no caching by default, and each `getaddrinfo` call will send an A & AAAA DNS request and take 5-50ms.

Reply:
> This turned out to be a false assumption. In Ubuntu there is no caching by default, and each getaddrinfo call sends an A & AAAA DNS request and takes 5-50ms.

On what are you basing this? On Ubuntu for:
```C#
[Benchmark]
public object GetHostEntry() => Dns.GetHostEntry("www.microsoft.com");
```
I get:
|         Method |     Mean |
|--------------- |---------:|
| GetHostEntry | 498.9 us |

Reply:
```C#
[Benchmark]
public void GetHostAddresses() => Dns.GetHostAddresses("fb.lt.main.ml.mdn.skype.net");
```

averages to `24.84 ms` in my local 22.04 environment and to `9.051 ms` in the Azure D32 VM from #92054.

Reply:
> `fb.lt.main.ml.mdn.skype.net`

That name resolves through a CNAME of `fb.lt.main.ml.mdn.trafficmanager.net`, which advertises a CNAME for `fileserver-backend-lb.lt-westus-01.ic3-middle-lane-main-lt.westus-test.cosmic-int.office.net` with a 0 second TTL. So my guess is the OS is just honoring the TTLs in this case.

The `www.microsoft.com` example above on the other hand uses non-0 TLLs.

Reply:
You can also check whether DNS caching in your system is enabled, e.g.
```sh
systemctl is-active systemd-resolved
```
My point is simply that stating that it's a false assumption that OSes cache DNS information is itself a false assumption. Every major OS I'm aware of is capable of doing DNS caching. Whether it's enabled is a different question, though to my knowledge it is by default in Ubuntu 22.04.

Reply:
I was wrong thinking that answers are never cached on Ubuntu by default. However, to me the comment suggests that we can generally rely on short completion times because of the caching, which is also not true, eg. as Miha pointed out answers with TTL=0 are not cached, which then increases the probability of hitting #81023.

I will add the comment back & augment it.